### ISSUE TITLE
Edge detection regex might be too loose

### ISSUE THREAD

**Issue by jodator** [ckeditor/ckeditor-dev#2207]
## Type of report

Bug

## Provide detailed reproduction steps (if any)

I think that Edge detection is not strict enough:

https://github.com/ckeditor/ckeditor-dev/blob/f9b46e125aa689ab5fa912574994e14211d5dc5a/core/env.js#L20

According to [MSDN article about Edge user agent string](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/compatibility/hh869301(v=vs.85)) the proper way to detect Edge would be:

```js
/edge\/(\d+.?\d*)/
```

tl;dr: I don't know why exactly space (`' '`) is allowed there and I think that only slash ('`/`') should be allowed there.

**Comment by StuartSilvester** [comment]
This is causing some Motorola Edge 5G devices to be identified as legacy Edge instead of Chrome/WebKit. The result is that CKEditor 4 cannot be loaded on these devices.

To replicate, set your browser user agent to the following and then visit any CKEditor 4 demo.
`Mozilla/5.0 (Linux; Android 12; motorola edge 5G UW (2021)) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Mobile Safari/537.36`
